node/pkg/p2p/registry_test.go

```diff
@@ -0,0 +1,60 @@
+package p2p
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	"github.com/mr-tron/base58"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestNewRegistry(t *testing.T) {
+	registry := NewRegistry()
+	assert.Equal(t, "", registry.guardianAddress)
+	assert.Equal(t, 0, len(registry.errorCounters))
+	assert.Equal(t, 0, len(registry.networkStats))
+}
+
+func TestSetGuardianAddress(t *testing.T) {
+	registry := NewRegistry()
+	assert.Equal(t, "", registry.guardianAddress)
+
+	registry.SetGuardianAddress("foo")
+	assert.Equal(t, "foo", registry.guardianAddress)
+}
+
+func TestSetNetworkStats(t *testing.T) {
+	registry := NewRegistry()
+	assert.Equal(t, 0, len(registry.networkStats))
+
+	pub, _, _ := ed25519.GenerateKey(rand.Reader)
+	contractAddr := base58.Encode(pub[:])
+
+	heart_beat := &gossipv1.Heartbeat_Network{
```

Comment:
As much as the pythonista in me hates it... It is idiomatic to [use mixedCase for variables names](https://go.dev/doc/effective_go#mixed-caps).

Writing idiomatic code for the language you're using makes maintenance easier down the road and especially for new developers added to the project.

Reply:
100%